file/rotator.go

```diff
@@ -245,13 +246,27 @@ func (r *Rotator) openNew() error {
 		return fmt.Errorf("failed to make directories for new file: %w", err)
 	}
 
-	_, err = os.Stat(r.rot.ActiveFile())
+	stat, err := os.Lstat(r.rot.ActiveFile())
 	if err == nil {
+		isSymlink := stat.Mode()&fs.ModeSymlink != 0
+
 		// check if the file has to be rotated before writing to it
 		reason, t := r.isRotationTriggered(0)
 		if reason == rotateReasonNoRotate {
-			return r.appendToFile()
+			// To avoid symlink following attacks, if the active file is a symlink
+			// we need to rotate it to avoid writing to the symlink target, which
+			// could be a sensitive or protected file not owned by us.
+			if isSymlink {
+				if r.log != nil {
+					r.log.Debugw("Active file is a symlink, forcing rotation", "filename", r.rot.ActiveFile())
```

Comment:
Making this a debug log means almost no one will see it, I would have preferred a warning I think, since this is an indicator something questionable has happened.